### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Dangelo-JAN**
# I created associations in my library app:

In this milestone I:

- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 require_relative 'person'
+require_relative 'classroom'
 
 class Studen < Person
+  attr_reader :classroom
+
   def initializate(age, classroom, name = 'Unknown', parent_permission: true)
     super(name, age, parent_permission)
     @classroom = classroom
```
Comment (uche-inyama) [inline student.rb]: Kindly correct the method name **initializate** to initialize.
However, if things have changed, kindly let me know.

 ## :slightly_smiling_face:

File: person.rb [hunk 2]
```diff
@@ -1,14 +1,16 @@
 require_relative 'nameable'
+require_relative 'rental'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, rentals
   attr_accessor :name, :age
 
   def initilizate(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
```
Comment (uche-inyama) [inline person.rb]: Kindly correct the method name **initializate** to initialize.

File: student.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 require_relative 'person'
+require_relative 'classroom'
 
 class Studen < Person
+  attr_reader :classroom
```
Comment (uche-inyama) [inline student.rb]: Kindly complete the name of the class.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Hi Dangelo-JAN :slightly_smiling_face:  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (Dangelo-JAN) [reply student.rb]: done

File: person.rb (hunk 2 above)
Comment (Dangelo-JAN) [reply person.rb]: done

File: student.rb (hunk 3 above)
Comment (Dangelo-JAN) [reply student.rb]: done

File: student.rb (hunk 1 above)
Comment (Whoistolu) [reply student.rb]: Well resolved 👍🏽

File: person.rb (hunk 2 above)
Comment (Whoistolu) [reply person.rb]: Well resolved 👍🏽

File: student.rb (hunk 3 above)
Comment (Whoistolu) [reply student.rb]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @Dangelo-JAN 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**